core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-favorite-panel/dot-pages-favorite-panel.component.spec.ts

```diff
@@ -0,0 +1,376 @@
+        modDate: '2020-09-02 16:45:15.569',
+        title: 'preview1',
+        screenshot: 'test1',
+        url: '/index1?host_id=A&language_id=1&device_inode=123',
+        owner: 'admin'
+    },
+    {
+        ...dotcmsContentletMock,
+        title: 'preview2',
+        modDate: '2020-09-02 16:45:15.569',
+        screenshot: 'test2',
+        url: '/index2',
+        owner: 'admin2'
+    }
+];
+
+describe('DotPagesFavoritePanelComponent', () => {
+    let fixture: ComponentFixture<DotPagesFavoritePanelComponent>;
+    let component: DotPagesFavoritePanelComponent;
+    let de: DebugElement;
+    let store: DotPageStore;
+    let dialogService: DialogService;
+
+    class storeMock {
+        get vm$() {
+            return of({
+                favoritePages: {
+                    items: [],
+                    showLoadMoreButton: false,
+                    total: 0
+                },
+                isEnterprise: true,
+                environments: true,
+                languages: [],
+                loggedUser: {
+                    id: 'admin',
+                    canRead: { contentlets: true, htmlPages: true },
+                    canWrite: { contentlets: true, htmlPages: true }
+                },
+                pages: {
+                    actionMenuDomId: '',
+                    items: [],
+                    addToBundleCTId: 'test1'
+                }
+            });
+        }
+    }
+
+    describe('Empty state', () => {
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [DotPagesFavoritePanelComponent, MockDotIconComponent],
+                imports: [
+                    BrowserAnimationsModule,
+                    DotMessagePipeModule,
+                    ButtonModule,
+                    DotPagesCardModule,
+                    DotPagesCardEmptyModule,
+                    PanelModule,
+                    HttpClientTestingModule
+                ],
+                providers: [
+                    DialogService,
+                    { provide: CoreWebService, useClass: CoreWebServiceMock },
+                    { provide: DotPageStore, useClass: storeMock },
+                    { provide: DotMessageService, useValue: messageServiceMock }
+                ]
+            }).compileComponents();
+        });
+
+        beforeEach(() => {
+            store = TestBed.inject(DotPageStore);
+            fixture = TestBed.createComponent(DotPagesFavoritePanelComponent);
+            de = fixture.debugElement;
+            component = fixture.componentInstance;
+
+            fixture.detectChanges();
+        });
+
+        it('should set panel with empty state class', () => {
+            const elem = de.query(By.css('p-panel'));
+            expect(
+                elem.nativeElement.classList.contains('dot-pages-panel__empty-state')
+            ).toBeTruthy();
+            expect(de.query(By.css('.dot-pages-panel-action__button'))).toBeFalsy();
```

Comment:
`data-testId`.

Reply:
removed cuz that assertion was not needed on this component